### PR TITLE
Refactor dunder init

### PR DESCRIPTION
**Pull request by waltsims**

### CODE REVIEW

**Review by faridyagubbayli — COMMENTED** [review]
Some comments

File: kwave/__init__.py [hunk 1]
```diff
@@ -11,80 +11,69 @@
...
-binary_path = os.path.join(Path(__file__).parent, "bin", system)
-environ["KWAVE_BINARY_PATH"] = binary_path
 
-url_base = "https://github.com/waltsims/"
-
-prefix = "https://github.com/waltsims/kspaceFirstOrder-{0}-{1}/releases/download/v1.3.0/"
+# Constants and Configurations
+URL_BASE = "https://github.com/waltsims/"
+BINARY_VERSION = "v1.3.0"
+PREFIX = f"{URL_BASE}kspaceFirstOrder-{{}}-{{}}/releases/download/{BINARY_VERSION}/"
```
[the remark this answers is not in the document]
Comment (waltsims) [reply kwave/__init__.py:18]: Not sure if it's good smell or where I got it from but allows you to evaluate the f-string twice. The first evaluation leaves the curly brackets in the f-string for the second evaluation step.

[the remark this answers is not in the document]
Comment (waltsims) [reply kwave/__init__.py:18]: image

File: kwave/__init__.py [hunk 2]
```diff
@@ -94,17 +83,17 @@ def _hash_file(filepath: str) -> str:
...
         return False
-    
+
     if binary_type is None:
         # this is non-kwave windows binary
         # it already exists according to the check above
         return True
-    
-    existing_metadata_path = os.path.join(binary_path, f'{binary_name}_metadata.json')
+    existing_metadata_path = os.path.join(BINARY_PATH, f"{binary_name}_metadata.json")
```
[the remark this answers is not in the document]
Comment (waltsims) [reply kwave/__init__.py:97]: done

[the remark this answers is not in the document]
Comment (waltsims) [reply kwave/__init__.py]: done

**Comment by waltsims** [comment]
Thanks for the comments! I've revisited things this evening. Have another look when you have a second.

File: kwave/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (faridyagubbayli) [reply kwave/__init__.py:18]: Hmm interesting, I never saw double braces in f-strings before and not sure if this is a good convention to follow or not.

**Review by faridyagubbayli — APPROVED** [review]
LGTM! Thanks for working on this!

**Pull request merged**